### ISSUE TITLE
Terminal.com closing, how to move Huginn?

### ISSUE THREAD

**Issue by alvarodemig** [cantino/huginn#1780]
I've been working with Huginn and Terminal.com for a year, but they are closing the service.
Now I need to get all my data from terminal and to install it in another service.
I'm absolutely lost with all this...
Any help?
Thanks

**Comment by cantino** [comment]
Hi @alvarodemig, they must have made it with a postgresql or mysql database. You should ask them if there's a way to export a database dump.

**Comment by alvarodemig** [comment]
Thanks @cantino ;)
I will try!
And... what is the EASIEST way to deploy huginn? I don't mind to pay (the less the better) or to use one of my pcs as a server. Maybe Heroku?
Thanks, I'm a mess in all this :(

**Comment by cantino** [comment]
Heroku is quite easy if you pay. There's a button on the README for it.

**Comment by alvarodemig** [comment]
Nothing from Terminal.com... I'll have to create all of them again by hand... :(

By the way, I'm trying to deploy it in Heroku but I can't.

I install the heroku toolbet (no .exe file? is it ok?)
I try to follow the instructions through the Windows command:
heroku --version
heroku login
And it seems ok, I'm logged.

I get the zip from https://github.com/cantino/huginn.git) and unzip it.
In the command I go to the Huginn directory and try:
cp .env.example .env && bundle
But it says that it does not recognise cp

------------------------------------------------------------

I try to do it through Heroku and I can't either. I try to use the "github", i found huginn and when I try to deploy the branchs it always fails.

--------------------------------------------------------------
Any help??

Thanks...

**Comment by cantino** [comment]
Hey @alvarodemig. Deploying Huginn is not supported on Windows.  We'd recommend using VirtualBox with Ubuntu (which you can do for free on Windows).  You could also use the Heroku button on our README to do a simple deployment to Heroku.

Re: terminal.com, one way to get your agents would be to add them all to a Scenario in Huginn, then export that Scenario as a JSON file. You'd be able to import that into a new Huginn instance later.

Also, if you can get an interactive terminal with terminal.com, you might be able to do a database dump. Do you know if they use mysql or postgresql?

**Comment by alvarodemig** [comment]
I had all my agents divided in Scenarios, I didn't know I could export them.
I stopped my Terminal instance to keep money to access when I had a solution, I've tried to resume it to try this scenario option, but it's been "resuming" for an hour... so... I think I just don't have access. They don't answer my mails. They are being really unhelpful... I think I will need just to recreate all of them by hand... I don't think anything changes with them after weeks trying.

I tried the Heroku button but it doesn't work for me. I choose an available name and US zone and I press "Deploy". One second later I get a "V" in "create app" and that's all. It doesn't continue and nothing changes.

This is what I get:
![image](https://cloud.githubusercontent.com/assets/14134694/20242616/957f477c-a8fa-11e6-8b29-b751f31416ac.png)

Should I do anything later?

If I can't fix it this way I´ll try the VirtualBox, although if I have it just on Heroku servers paying them is better for me. I just need the agents working.

And thank you very much @cantino . You're awesome

**Comment by cantino** [comment]
That's super strange... it just sits there? I just pushed the heroku button and it's working for me. Maybe you need to add a credit card to your Heroku account, or try a different browser?

To finish the Heroku setup, you will need to use Ubuntu or a Mac, so it's worth getting VirtualBox working either way.

**Comment by alvarodemig** [comment]
Hello again @cantino !

Fixed! I don't know why but today it just worked and continued... very weird... I had tried with different browsers and PCs... I'm in a paying plan, but Heroku told me that it should work with their free version.  I followed your advised and I installed VirtualBox and Ubuntu and now huginn is working! :D 

Anyway it's ok now

Thank you very much! I really appreciate what you do here.
